packages/safe-core-sdk/src/ethereumLibs/Web3Adapter.ts

```diff
@@ -125,13 +127,24 @@ class Web3Adapter implements EthAdapter {
     return this.#web3.eth.sign(message, this.#signerAddress)
   }
 
-  estimateGas(transaction: EthAdapterTransaction, options?: string): Promise<number> {
+  estimateGas(transaction: EthAdapterTransaction, options?: any): Promise<number> {
     return this.#web3.eth.estimateGas(transaction, options)
   }
 
   call(transaction: EthAdapterTransaction): Promise<string> {
     return this.#web3.eth.call(transaction)
   }
+
+  ensLookup(name: string): Promise<string> {
+    return this.#web3.eth.ens.getAddress(name)
+  }
+
+  async ensReverseLookup(address: string): Promise<string> {
+    const lookup = address.toLowerCase().substr(2) + '.addr.reverse'
+    const node = ethers.utils.namehash(lookup)
```

Comment:
ethers.js library should not be used in `Web3Adapter` class

Reply:
Unfortunately web3 doesn't provide a namehash implementation yet. We could instead include a new package [eth-ens-namehash](https://github.com/ensdomains/eth-ens-namehash) and use it in both places. What do you think?

Reply:
@usame-algan, have you checked to see if we use one in the Safe?

Reply:
It looks like this is the first place where we need the namehash for the reverse lookup. Ethers can do the reverse lookup without the namehash so this only affects web3 atm.

Reply:
@germartinez isn't ethers.js bundled with the Core SDK in any case? Or is there tree-shaking in place?

Reply:
`eth-ens-namehash` is very big too. I think it's better if use ethers temporarily. Web3 will include this function in one of the upcoming versions as per their GitHub.

Reply:
Let's not import ethers, but @ethersproject/hash sub-module instead (https://www.npmjs.com/package/@ethersproject/hash?activeTab=readme)

Reply:
Good Idea! I removed the `eth-ens-namehash` and used the sub-module instead.